store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReader.java

```diff
@@ -101,6 +108,43 @@ public static CarbonReaderBuilder builder(String tablePath, String tableName) {
     return reader.readSchema();
   }
 
+  /**
+   * Read carbonindex file and return the schema
+   * @param indexFilePath complete path including index file name
+   * @return null, if the index file is not present in the path.
```

Comment:
Can you optimize the expression? and can you add test case for returning null if the index file is not present in the path?

Reply:
Actually now it will not return null, It will throw exception. That why it is WIP. I will remove this from the method header.